src/Symfony/Component/Cache/Tests/DataCollector/CacheDataCollectorTest.php

```diff
@@ -0,0 +1,97 @@
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Cache\Adapter\TraceableAdapter;
+use Symfony\Component\Cache\DataCollector\CacheDataCollector;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+class CacheDataCollectorTest extends TestCase
+{
+    private const INSTANCE_NAME = 'test';
+
+    private $traceableAdapter;
+
+    public function testEmptyDataCollector()
+    {
+        $statistics = $this->getCacheDataCollectorStatisticsFromEvents([]);
+
+        $this->assertEquals($statistics[self::INSTANCE_NAME]['calls'], 0, 'calls');
+        $this->assertEquals($statistics[self::INSTANCE_NAME]['reads'], 0, 'reads');
+        $this->assertEquals($statistics[self::INSTANCE_NAME]['hits'], 0, 'hits');
+        $this->assertEquals($statistics[self::INSTANCE_NAME]['misses'], 0, 'misses');
+    }
+
+    public function testOneEventDataCollector()
+    {
+        $traceableAdapterEvent = new \StdClass();
+        $traceableAdapterEvent->name = 'getItem';
+        $traceableAdapterEvent->start = 0;
+        $traceableAdapterEvent->end = 0;
+        $traceableAdapterEvent->hits = 0;
+
+        $statistics = $this->getCacheDataCollectorStatisticsFromEvents([$traceableAdapterEvent]);
+
+        $this->assertEquals($statistics[self::INSTANCE_NAME]['calls'], 1, 'calls');
+        $this->assertEquals($statistics[self::INSTANCE_NAME]['reads'], 1, 'reads');
+        $this->assertEquals($statistics[self::INSTANCE_NAME]['hits'], 0, 'hits');
+        $this->assertEquals($statistics[self::INSTANCE_NAME]['misses'], 1, 'misses');
+    }
+
+    public function testHitedEventDataCollector()
+    {
+        $traceableAdapterEvent = new \StdClass();
+        $traceableAdapterEvent->name = 'hasItem';
+        $traceableAdapterEvent->start = 0;
+        $traceableAdapterEvent->end = 0;
+        $traceableAdapterEvent->hits = 1;
+        $traceableAdapterEvent->misses = 0;
+        $traceableAdapterEvent->result = ['foo' => false];
+
+        $statistics = $this->getCacheDataCollectorStatisticsFromEvents([$traceableAdapterEvent]);
+
+        $this->assertEquals($statistics[self::INSTANCE_NAME]['calls'], 1, 'calls');
+        $this->assertEquals($statistics[self::INSTANCE_NAME]['reads'], 1, 'reads');
+        $this->assertEquals($statistics[self::INSTANCE_NAME]['hits'], 1, 'hits');
+        $this->assertEquals($statistics[self::INSTANCE_NAME]['misses'], 0, 'misses');
+    }
+
+    public function testSavedEventDataCollector()
+    {
+        $traceableAdapterEvent = new \StdClass();
+        $traceableAdapterEvent->name = 'save';
+        $traceableAdapterEvent->start = 0;
+        $traceableAdapterEvent->end = 0;
+
+        $statistics = $this->getCacheDataCollectorStatisticsFromEvents([$traceableAdapterEvent]);
+
+        $this->assertEquals($statistics[self::INSTANCE_NAME]['calls'], 1, 'calls');
+        $this->assertEquals($statistics[self::INSTANCE_NAME]['writes'], 1, 'writes');
```

Comment:
Maybe make sure `writes` is zero on the other tests.

Reply:
Updated 👍